packages/live-preview-sdk/src/inspectorMode/index.ts

```diff
@@ -0,0 +1,212 @@
+
+      for (const eventTarget of eventTargets) {
+        const element = eventTarget as HTMLElement;
+        if (element.nodeName === 'BODY') break;
+        if (typeof element?.getAttribute !== 'function') continue;
+
+        if (this.handleTaggedElement(element)) {
+          return;
+        }
+      }
+
+      // Clear if no tagged element is hovered
+      if (this.hoveredElement) {
+        this.hoveredElement = undefined;
+        sendMessageToEditor(
+          InspectorModeEventMethods.MOUSE_MOVE,
+          { element: null },
+          this.targetOrigin
+        );
+      }
+    };
+
+    window.addEventListener('mouseover', onMouseOver);
+
+    return () => window.removeEventListener('mouseover', onMouseOver);
+  }
+
+  /** Sends scroll start and end event to the editor, on end it also sends the tagged elements again */
+  private bindScrollListener() {
+    const onScroll = () => {
+      if (!this.isScrolling) {
+        this.isScrolling = true;
+        sendMessageToEditor(InspectorModeEventMethods.SCROLL_START, {}, this.targetOrigin);
+      }
+
+      if (this.scrollTimeout) {
+        clearTimeout(this.scrollTimeout);
+      }
+
+      this.scrollTimeout = setTimeout(() => {
+        // No longer scrolling, let's update everything
+        this.isScrolling = false;
+        sendMessageToEditor(InspectorModeEventMethods.SCROLL_END, {}, this.targetOrigin);
+        this.sendAllElements();
+        if (this.hoveredElement) {
+          this.handleTaggedElement(this.hoveredElement);
+        }
+      }, 150);
+    };
+
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }
+
+  /** Detects DOM changes and sends the tagged elements to the editor */
+  private bindMutationListener() {
+    const mutationObserver = new MutationObserver(() => {
+      const taggedElements = getAllTaggedElements().filter(
+        (el) => !!getInspectorModeAttributes(el)
+      );
+
+      if (this.taggedElements?.length !== taggedElements.length) {
+        this.sendAllElements();
+      }
+    });
+
+    mutationObserver.observe(document.body, {
+      attributes: true,
+      attributeFilter: [
+        InspectorModeDataAttributes.ENTRY_ID,
+        InspectorModeDataAttributes.FIELD_ID,
+        InspectorModeDataAttributes.LOCALE,
+      ],
+      childList: true,
+      subtree: true,
+    });
+
+    return () => mutationObserver.disconnect();
+  }
+
+  /** Sends resize start and end event to the editor, on end it also sends the tagged elements again */
+  private bindResizeListener() {
+    const resizeObserver = new ResizeObserver(() => {
+      if (!this.isResizing) {
+        this.isScrolling = true;
+        sendMessageToEditor(InspectorModeEventMethods.RESIZE_START, {}, this.targetOrigin);
+      }
+
+      if (this.resizeTimeout) {
+        clearTimeout(this.resizeTimeout);
+      }
+
+      this.resizeTimeout = setTimeout(() => {
```

Comment:
Why are we using timeout here? for debounce / throttling?

Reply:
To get the information when the resizing (same for scrolling) ends and we can show the overlay in the live preview again.
During this interactions, the visuals would be not on the correct position

Reply:
Maybe we can add this as a small note? 👍

Reply:
should be already there now :) (same for scrolling)